daemon/daemon.go

```diff
@@ -345,16 +350,24 @@ func (daemon *Daemon) restore() error {
 
 			alive, _, process, err = daemon.containerd.Restore(context.Background(), c.ID, c.InitializeStdio)
 			if err != nil && !errdefs.IsNotFound(err) {
-				log.WithError(err).Error("failed to restore container with containerd")
+				logger(c).WithError(err).Error("failed to restore container with containerd")
 				return
 			}
-			if !alive && process != nil {
-				ec, exitedAt, err = process.Delete(context.Background())
-				if err != nil && !errdefs.IsNotFound(err) {
-					log.WithError(err).Error("failed to delete container from containerd")
-					return
+			logger(c).Debugf("alive: %v", alive)
+			if !alive {
+				// Process will only be nil here if `containerd.Restore` returned an errdefs.NotFoundError
+				// In this case we just want to cleanup a dead container from containerd.
```

Comment:
Ambiguous `in this case` the case we're in, or the case talked about in the line above? Will amend commit to clarify this.